earthfile2llb/interpreter.go

```diff
@@ -1631,6 +1652,75 @@ func (i *Interpreter) handleProject(ctx context.Context, cmd spec.Command) error
 	return nil
 }
 
+func (i *Interpreter) handlePipelineBlock(ctx context.Context, name string, block spec.Block) error {
+	if len(block) == 0 {
+		return errors.New("pipeline targets require sub-commands")
+	}
+
+	if block[0].Command == nil || block[0].Command.Name != "PIPELINE" {
+		return i.errorf(block[0].Command.SourceLocation, "PIPELINE must be the first command in a pipeline target")
+	}
+
+	for _, stmt := range block {
+		if stmt.Command == nil {
+			return errors.New("pipeline targets do not support IF, WITH, FOR, or WAIT commands")
+		}
+		cmd := *stmt.Command
+		var err error
+		switch cmd.Name {
+		case "PIPELINE":
+			err = i.handlePipeline(ctx, cmd)
+		case "TRIGGER":
+			err = i.handleTrigger(ctx, cmd)
+		case "ARG":
+			err = i.handleArg(ctx, cmd)
+		case "BUILD":
+			err = i.handleBuild(ctx, cmd, false)
+		default:
+			return i.errorf(cmd.SourceLocation, "pipeline targets only support PIPELINE, TRIGGER, ARG, and BUILD commands")
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (i *Interpreter) handlePipeline(ctx context.Context, cmd spec.Command) error {
+
+	if len(cmd.Args) > 1 {
+		return i.errorf(cmd.SourceLocation, "invalid number of PIPELINE arguments")
+	}
+
+	for _, arg := range cmd.Args {
+		if arg != "--push" {
+			return i.errorf(cmd.SourceLocation, "--push is the only supported PIPELINE argument")
+		}
+	}
```

Comment:
We should use arg parsing, via `parseArgs` the way we do it for the other commands, even if we only support one arg. There are some benefits, like code consistency and pretty error output. See https://github.com/earthly/earthly/blob/main/earthfile2llb/flags.go

Reply:
Cool. Sounds good.